state/execution.go

```diff
@@ -146,6 +146,13 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 		panic(err)
 	}
 	rawNewData := preparedProposal.GetBlockData()
+
+	// don't count the last tx in rpp.Txs which is data root back from app
```

Comment:
I think this comment is residual because AFAIK this branch doesn't have the data root in the last tx. It has a separate field for the data root hash:

```go
// Data contains all the information needed for a consensus full node to
// reconstruct an extended data square.
type Data struct {
	// Txs that will be applied to state in block.Height + 1 because deferred execution.
	// This means that the block.AppHash of this block does not include these txs.
	// NOTE: not all txs here are valid. We're just agreeing on the order first.
	Txs [][]byte `protobuf:"bytes,1,rep,name=txs,proto3" json:"txs,omitempty"`
	// SquareSize is the number of rows or columns in the original data square.
	SquareSize uint64 `protobuf:"varint,5,opt,name=square_size,json=squareSize,proto3" json:"square_size,omitempty"`
	// Hash is the root of a binary Merkle tree where the leaves of the tree are
	// the row and column roots of an extended data square. Hash is often referred
	// to as the "data root".
	Hash []byte `protobuf:"bytes,6,opt,name=hash,proto3" json:"hash,omitempty"`
}
```

so proposal to delete this line.

```suggestion
```

If it isn't residual, what is `rpp`?

Reply:
Yeah this would be residual from `main`